ir/main_test.go

```diff
@@ -0,0 +1,87 @@
+package ir_test
+
+import (
+	"fmt"
+
+	"github.com/llir/llvm/ir"
+	"github.com/llir/llvm/ir/constant"
+	"github.com/llir/llvm/ir/types"
+)
+
+func Example_main() {
+	// This example produces LLVM IR code equivalent to the following C code:
+	//
+	//    int main() {
+	//      int a = 32;
+	//      int b = 16;
+	//      return a + b;
+	//    }
+	//
+	// Read: https://blog.felixangell.com/an-introduction-to-llvm-in-go for inspiration
+	//
+	// Output:
+	//define i32 @main() {
+	//; <label>:0
+	//       %1 = alloca i32, align 4
+	//       %2 = alloca i32, align 4
+	//       %3 = alloca i32, align 4
+	//       store i32 0, i32* %1, align 4
+	//       store i32 32, i32* %2, align 4
+	//       store i32 16, i32* %3, align 4
+	//       %4 = load i32, i32* %2, align 4
+	//       %5 = load i32, i32* %3, align 4
+	//       %6 = add i32 %4, %5
+	//       ret i32 %6
+	//}
+
+	i32 := types.I32
+
+	// Create a new LLVM IR module.
+	mod := ir.NewModule()
+	//    int main() { ... }
+	main := mod.NewFunc("main", i32)
+
+	// Create an unnamed entry basic block and append it to the `main` function.
+	entry := main.NewBlock("")
+	// Create instructions and append them to the entry basic
+
+	// %1 = alloca i32, align 4
+	zero := entry.NewAlloca(i32)
+	zero.Align = 4
```

Comment:
Is `zero` ever used in other places of the IR module (it is written to through `store`, but never used)? If not, we can safely remove it without affecting the behaviour of the program. This would be in favour of making the introductory example as simple as possible.

However, people familiar with the output of Clang may prefer to see it written this way. Essentially `%1` holds the initial return value, I think? Later on we still return `tmpC` so the `zero` value is never used.